File: backend/tourism-service/src/main/java/com/example/tourismservice/entity/Beach.java
package com.example.tourismservice.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "beach")
public class Beach {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    @Column(name = "address")
    private String address;
    private String city;
    private String description;
    private double rating;
    @Column(name = "image_url")
    private String imageUrl;

    public Beach() {}

    public Beach(Long id, String name, String address, String city, String description, double rating, String image) {
        this.id = id;
        this.name = name;
        this.address = address;
        this.city = city;
        this.description = description;
        this.rating = rating;
        this.imageUrl = image;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }

    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public double getRating() { return rating; }
    public void setRating(double rating) { this.rating = rating; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
}

File: frontend/tourisme_app/lib/models/beach.dart
class Beach {
  final int id;
  final String name;
  final String address;
  final String city;
  final String description;
  final double rating;
  final String imageUrl;
  final double latitude;
  final double longitude;

  Beach({
    required this.id,
    required this.name,
    required this.address,
    required this.city,
    required this.description,
    required this.rating,
    required this.imageUrl,
    required this.latitude,
    required this.longitude,
  });

  factory Beach.fromJson(Map<String, dynamic> json) {
    return Beach(
      id: json['id'],
      name: json['name'],
      address: json['address'],
      city: json['city'] ?? '',
      description: json['description'] ?? '',
      rating: (json['rating'] ?? 0).toDouble(),
      imageUrl: json['imageUrl'] ?? '',
      latitude: (json['latitude'] ?? 0).toDouble(),
      longitude: (json['longitude'] ?? 0).toDouble(),
    );
  }
}